### ISSUE TITLE
Input and Output

### ISSUE THREAD

**Issue by Winnie09** [divyanshu-talwar/AutoImpute#6]
Dear author,

Thanks for developing the software! I have a few questions but did not find clear answer from your paper:

1. is the input a count matrix?
2. is the output a count matrix? Has it been normalized by library size, or log2-transformed?
3. I tried to use a raw count matrix as an input, but find negative values in output matrix?

Thanks!

**Comment by divyanshu-talwar** [comment]
Hey,

1. The input to `AutoImpute.py` is not a raw count matrix, but a pre-processed one. _You could check how to pre-process the raw count matrix in the `README` (Execution section) or in my reply for issue #5._ 
2. The python script `AutoImpute.py` assumes that the input matrix is a pre-processed one and being an autoencoder setup the output matrix mimics the same normalization properties. _(For more details on the pre-processing step check the `README`, the script for the same is `pre-process.R` resting in `AutoImpute\ Model/Pre-processing/` directory)._ 
3. Following the aforementioned steps should resolve your issue.

Also, as mentioned in my reply for issue #5, for best results it is advised to make sure that the hyper-parameters associated with the autoencoder model have been fine-tuned well enough by checking if the model's loss converges just fine.

Cheers!

**Comment by Winnie09** [comment]
OK. That’s helpful! Thanks!
On Mar 20, 2019, at 11:34 AM, Divyanshu Talwar <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey,

  1.  The input to AutoImpute.py is not a raw count matrix, but a pre-processed one. You could check how to pre-process the raw count matrix in the README (Execution section) or in my reply for issue #5<https://github.com/divyanshu-talwar/AutoImpute/issues/5>.
  2.  The python script AutoImpute.py assumes that the input matrix is a pre-processed one and being an autoencoder setup the output matrix mimics the same normalization properties. (For more details on the pre-processing step check the README, the script for the same is pre-process.R resting in AutoImpute\ Model/Pre-processing/ directory).
  3.  Following the aforementioned steps should resolve your issue.

Also, as mentioned in my reply for issue #5<https://github.com/divyanshu-talwar/AutoImpute/issues/5>, for best results it is advised to make sure that the hyper-parameters associated with the autoencoder model have been fine-tuned well enough by checking if the model's loss converges just fine.

Cheers!
